### PR TITLE
fix some warnings

### PR DESCRIPTION
**Pull request by ProkopRandacek**
They were spamming my build log :D

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
Thanks!

If you will, could you please indicate how to add these errors to the CI? Are you using a particular compiler?

On the CI, they are behind the `FTXUI_DEV_WARNINGS` option.

File: examples/dom/color_info_sorted_2d.ipp [hunk 1]
```diff
@@ -33,22 +32,22 @@ std::vector<std::vector<ftxui::ColorInfo>> ColorInfoSorted2D() {
...
           best_index = j;
         }
       }
-      std::swap(column[i], column[best_index]);
+      std::swap(column[i + 1], column[best_index]);
     }
   }
 
-  return info_columns;
+  return std::move(info_columns);
```
Comment (ProkopRandacek) [inline examples/dom/color_info_sorted_2d.ipp]: writing std move here disables copy elision and forces a call to the move constructor instead (as far as i know)

Comment (ArthurSonzogni) [reply examples/dom/color_info_sorted_2d.ipp]: You are right ;-)

Comment (ProkopRandacek) [reply examples/dom/color_info_sorted_2d.ipp]: then why did you revert it?

**Pull request merged**